src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -105,8 +105,13 @@ export const createSavedQueryService = (
   };
 
   const getSavedQuery = async (id: string): Promise<SavedQuery> => {
-    const savedObject = await savedObjectsClient.get<SerializedSavedQueryAttributes>('query', id);
-    if (savedObject.error) {
+    const {
+      saved_object: savedObject,
+      outcome,
+    } = await savedObjectsClient.resolve<SerializedSavedQueryAttributes>('query', id);
+    if (outcome === 'conflict') {
+      throw new Error(`Multiple saved queries found with ID: ${id}`);
```

Comment:
Can you change this error message to mention "legacy URL alias" so that a consumer can follow the bread crumbs and find the [docs](https://www.elastic.co/guide/en/kibana/master/legacy-url-aliases.html) to address the error?

Maybe something like this: 
```suggestion
      throw new Error(`Multiple saved queries found with ID: ${id} (legacy URL alias conflict)`);
```